### ISSUE TITLE
Exploit github api

### ISSUE THREAD

**Issue by jim-parry** [bcit-ci/codeigniter-website#44]
Now that we have a general purpose github api wrapper, it would be good to exploit that further, for instance extracting the number of downloads (CI3, CI2), total and over time. This might involve adding a table where we track the # of downloads at given points in time, so that we can report the usage over time (eg monthly), or acknowledge the most significant contributors in a month.

https://developer.github.com/v3/
https://developer.github.com/v3/repos/

**Comment by lonnieezell** [comment]
I had looked into this at one point for Bonfire and was unable to find anything satisfactory. The GitHub API didn't provide anything in v2 of the API. A quick glance at v3 and I don't see anything for download stats, either. I included a small script in the installer that would report back, but realized that every time the script got ran through on Travis or other CI instances where it needed to install the script to test it, it would give a false reading. 

The best place to get stats that I've seen is through packagist, but CI does more service through raw downloads then composer installs so that wouldn't be accurate. And if get the downloads from GitHub we'll need to combine it with Packagist installs, also, to get an accurate number. 

I like the "most significant/active contributors" aspect, though.

**Comment by jim-parry** [comment]
The releases API looks like the returned data could be parsed, to infer some download data. Github used to have a downloads API, but it has been deprecated :(

Acknowledging our active community members does seem like a win-win. It might be possible to extract something similar from the forum. Hmmm. Let's see if anyone jumps on this :)

**Comment by michalsn** [comment]
I had some time to take a look at this…

Sorry to say that, but I was unable to get info about releases, and when I was trying to do so with other repos, the assets array was always empty... so for now there is no way to get number of downloads for releases :(

**Comment by jim-parry** [comment]
I figured out the contributors part, and submitted a PR for that.

**Comment by zhanghongyi** [comment]
Would you mind me jumping into the thread? :)

For the download part, I am curious that if it would be easier and more accurate to count directly from the official website. I think most of the users are downloading from the official website rather than Github (at least me...). The idea of exploiting github api is cool though.

**Comment by jim-parry** [comment]
No ... the official site links to github for downloads, so they would have th emost accurate count, if we had access to it.

**Comment by zhanghongyi** [comment]
@jim-parry ah I see. Now it totally makes sense to me. Thanks.

**Comment by zhanghongyi** [comment]
I explored it a bit further. 

As @michalsn pointed out, when we tried to get release infos of CodeIgniter, the return is always empty. This issues has been discussed in this thread [https://github.com/bcit-ci/CodeIgniter/issues/3421]. The reason is that we are not supporting the full release but only tags. 

For tags, github api does not have any support for downloading counts. But if we do have full releases that can be accessed through `GET /releases`, we should be able to have some progress since for every single release, there is a field `"download_count"` in the assets of the returned json. Refer to the official guide: https://developer.github.com/v3/repos/releases/#get-a-single-release
